cmd/ndm_daemonset/probe/eventhandler.go

```diff
@@ -63,32 +107,40 @@ func (pe *ProbeEvent) addBlockDeviceEvent(msg controller.EventMessage) {
 
 // deleteBlockDeviceEvent deactivate blockdevice resource using uuid from etcd
 func (pe *ProbeEvent) deleteBlockDeviceEvent(msg controller.EventMessage) {
-	blockDeviceOk := pe.deleteBlockDevice(msg)
-
-	// when one disk is removed from node and entry related to
-	// that disk is not present in etcd,  in that case it
-	// again rescan full system and update etcd accordingly.
-	if !blockDeviceOk {
-		go pe.initOrErrorEvent()
-	}
-}
-
-func (pe *ProbeEvent) deleteBlockDevice(msg controller.EventMessage) bool {
 	bdList, err := pe.Controller.ListBlockDeviceResource()
 	if err != nil {
 		klog.Error(err)
-		return false
 	}
-	ok := true
-	for _, diskDetails := range msg.Devices {
-		existingBlockDeviceResource := pe.Controller.GetExistingBlockDeviceResource(bdList, diskDetails.UUID)
+
+	isDeactivated := true
+	isGPTBasedUUIDEnabled := pe.Controller.FeatureGates.IsEnabled(features.GPTBasedUUID)
+
+	for _, device := range msg.Devices {
+		// create the new UUID for removing the device
+		if isGPTBasedUUIDEnabled {
+			uuid, ok := generateUUID(*device)
+			if !ok {
+				klog.Info("could not recreate UUID while removing device")
+				continue
+			}
+			// use the new UUID for deactivating the blockdevice
+			device.UUID = uuid
+		}
+
+		existingBlockDeviceResource := pe.Controller.GetExistingBlockDeviceResource(bdList, device.UUID)
 		if existingBlockDeviceResource == nil {
-			ok = false
+			isDeactivated = false
 			continue
 		}
 		pe.Controller.DeactivateBlockDevice(*existingBlockDeviceResource)
 	}
-	return ok
+
+	// TODO @akhilerm check if rescan is required every time and take appropriate actions
+	//
+	// rescan only if GPT based UUID is disabled.
+	if !isDeactivated && !isGPTBasedUUIDEnabled {
```

Comment:
@pawanpraka1  I also have a concern or rescanning everytime, when a device which was not used is removed. Even in case of filtered out devices. This can happen. IMO we should avoid these also. 
Need your thoughts on why it was introduced in the first place.

Reply:
yes, the idea was if any thing fails, scan the entire disks. For filtered disks, this was probably a bug as we look for that device in etcd, which will not be there so we scan it again unnecessary.

Reply:
ok. then will update the code so that a rescan is not performed, if there is any error during `deleteBlockDeviceEvent`

Reply:
done.